### PR TITLE
Extend processor support to include AMD + a few Intel platforms

### PR DESCRIPTION
**Pull request by Zormeister**
Very much a draft as this hasn't been tested on real hardware

### CODE REVIEW

**Comment by Zormeister** [comment]
Can't build locally like how GitHub actions does, `libSystem_kernel.a` fails at the linking stage

macOS Version: Sonoma 14.2.1, `23C71`
Xcode version: 15.1, `15C65`

Build output:
`
[688/692] Linking C static library src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(mach_approximate_time.s.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(mach_bridge_remote_time.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(varargs_wrappers.s.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(accept.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(bind.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(connect.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(getattrlist.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(getpeername.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(getsockname.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(kill.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(lchown.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(listen.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(mprotect.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(msync.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(munmap.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(recvfrom.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(recvmsg.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(select.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(sendmsg.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(sendto.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(setattrlist.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(sigsuspend.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(socketpair.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(mach_approximate_time.s.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(mach_bridge_remote_time.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(varargs_wrappers.s.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(accept.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(bind.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(connect.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(getattrlist.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(getpeername.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(getsockname.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(kill.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(lchown.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(listen.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(mprotect.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(msync.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(munmap.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(recvfrom.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(recvmsg.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(select.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(sendmsg.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(sendto.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(setattrlist.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(sigsuspend.c.o) has no symbols
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.a(socketpair.c.o) has no symbols
[689/692] Linking C shared library src/Libraries/libSystem/libsystem_kernel/libsystem_kernel.dylib
FAILED: src/Kernel/xnu/xnu_build/src/xnu-stamp/xnu-build <PureDarwin Source Directory>/build/src/Kernel/xnu/xnu_build/src/xnu-stamp/xnu-build`

**Comment by wjk** [comment]
@Zormeister I haven’t booted PureDarwin ever, including inside a VM, since I don’t have a working bootloader. If you have one, please send it to me so I can integrate it into the source code.

File: src/Kernel/xnu/osfmk/i386/cpuid.h [hunk 1]
```diff
@@ -61,6 +61,13 @@
 #define _Bit(n)                 (1ULL << n)
 #define _HBit(n)                (1ULL << ((n)+32))
 
+/*
+ * This is so we don't have a global isAMD variable.
+ */
+#define CPUID_VEN_UNKNWON       0
```
Comment (ninjadev64) [inline src/Kernel/xnu/osfmk/i386/cpuid.h:67]: Also, typo here

**Comment by Zormeister** [comment]
@wjk No bootloader, I tried running it through `kmutil` but it errors out trying to build the KC, AFAICT it’s something about the way the kernel is built, quote `“missing split seg v2”`

File: src/Kernel/xnu/osfmk/i386/cpuid.h (hunk 1 above)
Comment (Zormeister) [reply src/Kernel/xnu/osfmk/i386/cpuid.h:67]: Will fix up soon, I have a local commit that hasn’t been pushed yet

**Comment by Zormeister** [comment]
oh shoot

**Comment by ninjadev64** [comment]
What

**Pull request closed without merging**